### PR TITLE
Allow setting of git ref on start step

### PR DESCRIPTION
**Pull request by codyleblanc**
This should address #4 #5. I ran into the same issue when deploying my Pull Requests for my team.

Let me know if I missed anything or if it's completely off the mark!

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -20,20 +20,22 @@ async function run() {
       {
         const environment = core.getInput('env', { required: true });
         const noOverride = core.getInput('no_override') === 'true';
-        let deploymentID = core.getInput('deployment_id', { required: false });
-        console.log(`initializing deployment ${deploymentID} for ${environment}`);
+        const transient = core.getInput('transient') === 'true';
+        const git_ref = core.getInput('ref') || ref;
```
Comment (bobheadxi) [inline src/main.ts:24]: a nit for consistency
```suggestion
        const gitRef = core.getInput('ref') || ref;
```

```suggestion
        const gitRef = core.getInput('ref') || ref;
```

File: src/main.ts [hunk 2]
```diff
@@ -20,20 +20,22 @@ async function run() {
...
           await deactivateEnvironment(client, repo, environment);
         }
 
-        const transient = core.getInput('transient', { required: false }) === 'true';
         if (!deploymentID) {
           const deployment = await client.repos.createDeployment({
             owner: repo.owner,
             repo: repo.repo,
-            ref: ref,
+            ref: git_ref,
```
Comment (bobheadxi) [inline src/main.ts:38]:
```suggestion
            ref: gitRef,
```

File: src/main.ts [hunk 3]
```diff
@@ -42,10 +44,10 @@ async function run() {
           deploymentID = deployment.data.id.toString();
         }
 
-        console.log(`created deployment ${deploymentID} for env ${environment}`);
+        console.log(`created deployment ${deploymentID} for ${environment} @ ${git_ref}`);
```
Comment (bobheadxi) [inline src/main.ts:47]:
```suggestion
        console.log(`created deployment ${deploymentID} for ${environment} @ ${gitRef}`);
```

File: README.md [hunk 4]
```diff
@@ -111,6 +112,35 @@ jobs:
 
 <br />
 
+<details>
+<summary>Simple Pull Request Example</summary>
+<p>
+
+```yml
+on:
+  pull_request:
```
[the remark this answers is not in the document]
Comment (codyleblanc) [reply README.md:121]: Hey @bobheadxi, I haven't run into any issues with the deployments not being re-run with just the `pull_request` event type. I believe the `synchronization` event encompasses pushing to the branch that the pull request is based on.

It's hard to find documentation on it but I did find a reference in this [blog post](https://github.blog/2011-10-19-all-of-the-hooks/)

> `pull_request` – Any time a Pull Request is opened, closed, or synchronized (updated due to a new push in the branch that the pull request is tracking).

File: src/main.ts (hunk 1 above)
Comment (codyleblanc) [reply src/main.ts:24]: 🤦 I was switching back and forth between this and Ruby. Thanks for catching it.

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (bobheadxi) [reply README.md:121]: Ah great, TIL! Thanks 👍

**Pull request merged**